packages/react/src/components/Tag/SelectableTag.tsx

```diff
@@ -101,6 +102,12 @@ const SelectableTag = forwardRef(
     });
     const [isEllipsisApplied, setIsEllipsisApplied] = useState(false);
 
+    useEffect(() => {
+      if (selected !== selectedTag) {
+        setSelectedTag(selected);
+      }
+    }, [selected]);
```

Comment:
Does this need an effect? 

https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

```suggestion
    if (selected !== selectedTag) {
      setSelectedTag(selected);
    }
```

Reply:
I think we still need useEffect here. `selected` and `onChange`  are optional props to the component, so if parent component not passing `selected` prop Tag component should handle is inside own state.

Do you agree? 🤔

Reply:
@tay1orjones  Please have a look

Reply:
@sojinantony01 I forgot we have the `useControllableState` hook you can use for this.

https://github.com/carbon-design-system/carbon/blob/15b90a42ed5d35b269396669f3dd72328b58b8dc/packages/react/src/internal/useControllableState.ts#L38

It'll handle the whole thing for you. We'll need to add a `defaultSelected` boolean prop to SelectableTag to pass to the hook. Probably the closest usage example you could follow is in Toggle:

https://github.com/carbon-design-system/carbon/blob/a6eb26586b0531ff65150398adebdee7cdd6e4e2/packages/react/src/components/Toggle/Toggle.tsx#L121-L125

We'll want to make sure there's clear passing tests for both controlled and uncontrolled modes here too.

Reply:
Thats really a useful hook. Thanks @tay1orjones.
I have updated the component using the `useControllableState` hook. we already have tests for uncontrolled state and i have added couple of tests for the controlled case - All passing.

And have manually tested the component using storybook.
Please have a look in the latest code.